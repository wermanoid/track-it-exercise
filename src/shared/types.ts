import React from 'react';

export type Omit<T, K> = Pick<T, Exclude<keyof T, K>>;

export enum IssueState {
  OPEN = 'Open',
  PENDING = 'Pending',
  CLOSED = 'Closed',
}

export interface Issue {
  id: string;
  title: string;
  description: string | null;
  created: Date | string;
  status: IssueState;
}

export interface IssueStatus {
  status: IssueState;
  label: string;
  icon: React.ComponentType<{}>;
}

export type CreateIssueArgs = Pick<Issue, 'title'> &
  Partial<Pick<Issue, 'description'>>;

export type UpdateIssueArgs = Pick<Issue, 'id'> &
  Partial<Pick<Issue, 'description' | 'title' | 'status'>>;
