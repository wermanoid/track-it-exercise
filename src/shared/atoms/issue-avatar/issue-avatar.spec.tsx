import React from 'react';
import { mount } from 'enzyme';

import { IssueState } from '#shared/types';
import IssueAvatar from './issue-avatar';

// TODO: Check why .toHaveStyleRule is not added to expect

describe('issue-avatar:', () => {
  const content = 'C';

  it('should render with #eee background', () => {
    const wrap = mount(<IssueAvatar>{content}</IssueAvatar>);

    expect(wrap).toMatchSnapshot();
  });

  it('should render with #red background', () => {
    const wrap = mount(
      <IssueAvatar status={IssueState.OPEN}>{content}</IssueAvatar>
    );

    expect(wrap).toMatchSnapshot();
  });

  it('should render with #darkorange background', () => {
    const wrap = mount(
      <IssueAvatar status={IssueState.PENDING}>{content}</IssueAvatar>
    );

    expect(wrap).toMatchSnapshot();
  });

  it('should render with #green background', () => {
    const wrap = mount(
      <IssueAvatar status={IssueState.CLOSED}>{content}</IssueAvatar>
    );

    expect(wrap).toMatchSnapshot();
  });
});
