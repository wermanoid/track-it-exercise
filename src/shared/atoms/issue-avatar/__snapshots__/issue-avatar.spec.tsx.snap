// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`issue-avatar: should render with #darkorange background 1`] = `
.emotion-0.emotion-0 {
  background-color: darkorange;
}

<IssueAvatar
  status="Pending"
>
  <WithStyles(Avatar)
    className="emotion-0 emotion-1"
    status="Pending"
  >
    <Avatar
      className="emotion-0 emotion-1"
      classes={
        Object {
          "colorDefault": "MuiAvatar-colorDefault-2",
          "img": "MuiAvatar-img-3",
          "root": "MuiAvatar-root-1",
        }
      }
      component="div"
      status="Pending"
    >
      <div
        className="MuiAvatar-root-1 MuiAvatar-colorDefault-2 emotion-0 emotion-1"
        status="Pending"
      >
        C
      </div>
    </Avatar>
  </WithStyles(Avatar)>
</IssueAvatar>
`;

exports[`issue-avatar: should render with #eee background 1`] = `
.emotion-0.emotion-0 {
  background-color: #eee;
}

<IssueAvatar>
  <WithStyles(Avatar)
    className="emotion-0 emotion-1"
  >
    <Avatar
      className="emotion-0 emotion-1"
      classes={
        Object {
          "colorDefault": "MuiAvatar-colorDefault-2",
          "img": "MuiAvatar-img-3",
          "root": "MuiAvatar-root-1",
        }
      }
      component="div"
    >
      <div
        className="MuiAvatar-root-1 MuiAvatar-colorDefault-2 emotion-0 emotion-1"
      >
        C
      </div>
    </Avatar>
  </WithStyles(Avatar)>
</IssueAvatar>
`;

exports[`issue-avatar: should render with #green background 1`] = `
.emotion-0.emotion-0 {
  background-color: green;
}

<IssueAvatar
  status="Closed"
>
  <WithStyles(Avatar)
    className="emotion-0 emotion-1"
    status="Closed"
  >
    <Avatar
      className="emotion-0 emotion-1"
      classes={
        Object {
          "colorDefault": "MuiAvatar-colorDefault-2",
          "img": "MuiAvatar-img-3",
          "root": "MuiAvatar-root-1",
        }
      }
      component="div"
      status="Closed"
    >
      <div
        className="MuiAvatar-root-1 MuiAvatar-colorDefault-2 emotion-0 emotion-1"
        status="Closed"
      >
        C
      </div>
    </Avatar>
  </WithStyles(Avatar)>
</IssueAvatar>
`;

exports[`issue-avatar: should render with #red background 1`] = `
.emotion-0.emotion-0 {
  background-color: red;
}

<IssueAvatar
  status="Open"
>
  <WithStyles(Avatar)
    className="emotion-0 emotion-1"
    status="Open"
  >
    <Avatar
      className="emotion-0 emotion-1"
      classes={
        Object {
          "colorDefault": "MuiAvatar-colorDefault-2",
          "img": "MuiAvatar-img-3",
          "root": "MuiAvatar-root-1",
        }
      }
      component="div"
      status="Open"
    >
      <div
        className="MuiAvatar-root-1 MuiAvatar-colorDefault-2 emotion-0 emotion-1"
        status="Open"
      >
        C
      </div>
    </Avatar>
  </WithStyles(Avatar)>
</IssueAvatar>
`;
